### PR TITLE
Column visibility and FeatureListFX column menu

### PR DESCRIPTION
**Pull request by robinschmid**
Still in progress.

### Done
- Better context menu on the + button in the right corner of the table
- Short cut to only show small table or all columns

### TODO
- [x] Fixed order for columns (ID, RT, MZ, Height, Area, .... Shape, Feature: State, RT, MZ, Height, Area, .... )
- [x] Control visibility for all columns and sub columns (Spectral match:Compound, RTRange:min ...)

### CODE REVIEW

File: src/main/java/io/github/mzmine/datamodel/features/types/fx/ColumnID.java [hunk 1]
```diff
@@ -71,9 +78,7 @@ public boolean equals(Object o) {
       return false;
     }
     ColumnID columnID = (ColumnID) o;
-    return type == columnID.type &&
-        dt.equals(columnID.dt) &&
-        Objects.equals(raw, columnID.raw);
+    return type == columnID.type && dt.equals(columnID.dt) && Objects.equals(raw, columnID.raw);
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/datamodel/features/types/fx/ColumnID.java:81]: dont forget the sub column index

File: src/main/java/io/github/mzmine/datamodel/features/types/DataTypes.java [hunk 2]
```diff
@@ -30,17 +33,43 @@
 public class DataTypes {
 
   private static final Logger logger = Logger.getLogger(DataTypes.class.getName());
-  public static ConcurrentHashMap<Class<? extends DataType>, DataType> TYPES = new ConcurrentHashMap<>();
+
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/datamodel/features/types/DataTypes.java:36]: Looks like io.github.mzmine.modules.io.projectload.version_3_0.DataTypes,java to me. Should we just reuse that class then?

Comment (SteffenHeu) [reply src/main/java/io/github/mzmine/datamodel/features/types/DataTypes.java:36]: ah ok you deleted my class and made it yours :D

File: src/main/java/io/github/mzmine/modules/io/projectload/version_3_0/FeatureListLoadTask.java [hunk 3]
```diff
@@ -103,25 +105,25 @@ public static Object parseDataType(XMLStreamReader reader, DataType<?> type,
...
     return "Importing feature list " + currentFlist + (processedFlists + 1) + "/" + numFlists
-        + ". Parsing row " + processedRows + "/" + totalRows;
+           + ". Parsing row " + processedRows + "/" + totalRows;
   }
 
   @Override
   public double getFinishedPercentage() {
     return (double) processedFlists / numFlists + ((double) 1 / numFlists) * ((double) processedRows
-        / totalRows);
+                                                                              / totalRows);
```
Comment (SteffenHeu) [inline src/main/java/io/github/mzmine/modules/io/projectload/version_3_0/FeatureListLoadTask.java:126]: this format is horrible ^^

**Comment by SteffenHeu** [comment]
The stuff noone likes to do, thanks Robin!

File: src/main/java/io/github/mzmine/datamodel/features/types/DataTypes.java (hunk 2 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/datamodel/features/types/DataTypes.java:36]: yes 🗡️

File: src/main/java/io/github/mzmine/modules/io/projectload/version_3_0/FeatureListLoadTask.java (hunk 3 above)
Comment (robinschmid) [reply src/main/java/io/github/mzmine/modules/io/projectload/version_3_0/FeatureListLoadTask.java]: This is your doing :D
and its only autoformatted.... 
I changed it now.

**Pull request merged**